generator/docs/Contradictions.md

```diff
@@ -0,0 +1,51 @@
+# Contradictions
+
+Contradictions are where constraints are combined in some fashion, creating a representation of data where no, or incorrectly reduced, amounts of data can be produced. The following categories of contradictions can occur in a profile:
+
+This document describes [how data is generated](SetRestrictionAndGeneration.md) and underpins the concept of contradictions.
+
+| Type | Explanation | Example |
+| ---- | ---- | ---- |
+| 'Hard' | _cannot_ create any rows of data (for the field, and therefore the output file) | `foo ofType string` and `foo ofType numeric` and `foo not(is null)` - no rows would be emitted |
+| 'Soft' | _could_ create some data, but some scenarios would produce none | `foo not null` and `if bar equalTo 1 then foo is null`, also `foo ofType string` and `foo ofType numeric`, 1 row would be emitted, containing `null` |
+
+There is an optional component of the generator - the profile validator - which can detect some of the above contradictions. See the explanation of each type for more detail.
+
+## 'Hard' contradictions
+This is where contradictions occur in a way where no rows could be satisfied for at least one field. If this is the case then no rows can be emitted for any field in the profile, therefore the output file/s would be empty.
+Hard contradictions can otherwise be described as removed all possible values from the universal set and denying the absence of values for the field; `not(is null)`. All hard contradictions must contain this constraint in some fashion, otherwise `null` can still be emitted.
+
```

Comment:
This sentence mixes tenses I think, perhaps "Hard contradictions can otherwise be described as *removing* all possible values from the universal set and denying the absence of values for the field" would be better?

Reply:
Good spot, updated